src/components/DatePicker/Calendar.tsx

```diff
@@ -0,0 +1,297 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  SvgIcChevronDown,
+  SvgIcChevronLeft,
+  SvgIcChevronRight,
+  SvgIcChevronUp,
+} from "../../assets/svg-components";
+import "./DatePicker.scss";
+import { months, years } from "../../utils/dateHandler";
+export interface CalendarProps {
+  dateVal?: string;
+  onDateClick: Function;
+  isRange?: boolean;
+  from: string;
+  to: string;
+  calendar: any;
+  selectedMonth: string;
+  selectedYear: string;
+  monthHandler: Function;
+  yearHandler: Function;
+  handleNextButtonClicked?: Function;
+  handlePreviousButtonClicked?: Function;
+  calendarId: string;
+}
+
+const Calendar = (props: CalendarProps) => {
+  const {
+    dateVal,
+    onDateClick,
+    isRange,
+    from,
+    to,
+    calendar,
+    selectedMonth,
+    selectedYear,
+    monthHandler,
+    yearHandler,
+    handleNextButtonClicked,
+    handlePreviousButtonClicked,
+    calendarId,
+  } = props;
+
+  const [days, setDays] = useState([
+    { name: "S", enum: 7 },
+    { name: "M", enum: 1 },
+    { name: "T", enum: 2 },
+    { name: "W", enum: 3 },
+    { name: "T", enum: 4 },
+    { name: "F", enum: 5 },
+    { name: "S", enum: 6 },
+  ]);
+  useEffect(() => {}, [props]);
+
+  const [showMonthToggle, setShowMonthToggle] = useState(false);
+  const [showYearToggle, setShowYearToggle] = useState(false);
+
+  const handleMonthClicked = (monthIndex: number) => {
+    monthHandler(monthIndex);
+    setShowMonthToggle(false);
+  };
+
+  const handleYearClicked = (year: any) => {
+    yearHandler(year);
+    setShowYearToggle(false);
+  };
+
+  const handleDateClick = (calendarItem: any) => {
+    let dd =
```

Comment:
It appears that the value of dd is not being modified anywhere in the function. Therefore, you can consider using const instead of let to declare this variable.

Reply:
Fixed